### PR TITLE
#2137 flatten email/pdf reports

### LINKED ISSUE #2137: PDF/HTML Reports: Expand Actions reports (and some others) by default
It would be sometimes useful to Expand (show recursively) the data for some reports since the expanded data is very important for these:
- Page Urls, Page Titles, Outlinks, downloads
- [Custom variables name /values](http://piwik.org/docs/custom-variables/)
- [Campaign names/keyword](http://piwik.org/docs/tracking-campaigns/)

When creating the Scheduled report, users could check a box "For Pages, Downloads, Outlinks report, Expand and show Top Pages in the hierarchy"
- if a category has 10 children categories and pages, I propose we display the 4 top ones in each sub category.
- The 5th listing will be the truncated row ("Others") or the 5th page/category if there were only 5.
  - countries/fr/paris/
  - countries/fr/marseille/
  - countries/fr/lyon/
  - countries/fr/bordeaux/
  - countries/fr/Others (this is the truncated row)

See  #2816 and #2742

This was reported in the forum [this post](http://forum.piwik.org/read.php?2,72844) and [here](http://forum.piwik.org/read.php?3,83855)

### PR DESCRIPTION
**Pull request by cbuchli**
Adds a new Flag to Scheduled Reports that makes it possible to flatten the generated reports.

The behaviour is 100% backwards compatible and doesn't need any update-scripts to be run since the default is set to false and the flag isn't marked as mandatory. It's stored in the report's parameters field and thus doesn't change the DB-Scheme neither.

Resolves #2137 in a "quickfix" manner.

Feature supported by [EDA Switzerland](https://www.eda.admin.ch/)

### CODE REVIEW

**Comment by cbuchli** [comment]
Pull Request WIP

**Comment by cbuchli** [comment]
@mattab the 'Pull Request WIP' can be removed: I've tested my requirements and removed all side-effects. The Feature is also included in the UI-Test (new Checkbox "Flatten" in addReport mask).

I'll stand by to fix and adjust anything you find during review.

File: plugins/ScheduledReports/templates/_addReport.twig [hunk 1]
```diff
@@ -93,6 +93,13 @@
                  title="{{ 'ScheduledReports_EvolutionGraph'|translate(5)|e('html_attr') }}">
             </div>
         </div>
+        <div ng-show="manageScheduledReport.report.type == 'email'">
+            <div piwik-field uicontrol="checkbox"
+                 name="display_flat"
+                 ng-model="manageScheduledReport.report.displayFlat"
+                 title="{{ 'General_Flatten'|translate|e('html_attr') }}">
```
Comment (tsteur) [inline plugins/ScheduledReports/templates/_addReport.twig]: I wonder if it is enough to show "Flatten" or whether could explain a little. Like Flatten will be only applied to reports that support it, and by default it is shown "hierarchical". Although I think in reports it is not really shown hierarchical by default but only the first level of the report which is not really useful so flatten is quite valuable. I think we need to explain it to users. @mattab  maybe an idea?

**Comment by tsteur** [comment]
Worked for me and is useful 👍  Only need to work a little on the wording in the UI I'd say

**Comment by Fensterbank** [comment]
I'm looking forward to see this merged. 👍

**Comment by mattab** [comment]
After testing UI and thinking about all reports that support Flatten, now it feels like we should keep things even more simple while trying to be useful to most people...so i'd go even further and enable flatten for all reports by default, and not make it configurable. So far i could not think of a use case where you'd want the reports hierarchical... can't believe it took us so long to address this! if the need to have it configurable would come up, we could re-use your PR. I'll open a separate PR and close this one. thanks for creating it and hope you will check out some other issues :+1:

**Pull request closed without merging**